ext/standard/formatted_print.c

```diff
@@ -634,15 +634,14 @@ php_formatted_print(char *format, size_t format_len, zval *args, int argc, int n
 
 				case 'd':
 					php_sprintf_appendint(&result, &outpos,
-										  zval_get_long(tmp),
-										  width, padding, alignment,
-										  always_sign);
+						zval_get_long_ex(tmp, /* is_strict */ true),
```

Comment:
I'm not sure about these changes. I could see an argument being made that `%d` should be considered an explicit int cast.

Reply:
Well `%s` does emit a warning when you pass in an array so I thought throwing the deprecation notice would be somewhat consistent (https://3v4l.org/hfbbX)

Reply:
This warning comes from the explicit cast: https://3v4l.org/Frq47

Reply:
TIL